### PR TITLE
Store eligibility proof in block and validate it

### PR DESCRIPTION
**Pull request by noamnelke**

### CODE REVIEW

File: cmd/node/node.go [hunk 1]
```diff
@@ -374,19 +374,20 @@ func (app *SpacemeshApp) Start(cmd *cobra.Command, args []string) {
 
 	app.unregisterOracle = func() { oracleClient.Unregister(true, pub.String()) }
 
-	//bo := oracle.NewBlockOracleFromClient(oracleClient, int(app.Config.CONSENSUS.NodesPerLayer))
 	nodesPerLayer := app.Config.CONSENSUS.NodesPerLayer
 	layersPerEpoch := app.Config.CONSENSUS.LayersPerEpoch
 	activationDb := &activation.ActivationDb{}      // TODO: initialize properly
```
Comment (almogdepaz) [inline cmd/node/node.go:388]: app.Config.CONSENSUS.NodesPerLayer is the size of the layer ? 
if so we already have a flag for that

File: miner/block_builder.go [hunk 2]
```diff
@@ -246,19 +248,21 @@ func (t *BlockBuilder) acceptBlockData() {
...
 				}
-				t.network.Broadcast(meshSync.NewBlockProtocol, bytes)
-			}()
+				go func() {
+					bytes, err := mesh.BlockAsBytes(*blk)
+					if err != nil {
+						t.Log.Error("cannot serialize block %v", err)
+						return
+					}
+					t.network.Broadcast(meshSync.NewBlockProtocol, bytes)
```
Comment (almogdepaz) [inline miner/block_builder.go]: why is NewBlockProtocol defined in sync? 
move it here

File: cmd/node/node.go (hunk 1 above)
Comment (noamnelke) [reply cmd/node/node.go:388]: As Shaggy would put it: it wasn't me...
https://github.com/spacemeshos/go-spacemesh/commit/fb3af280e26293d1e2c86463798e6817dba1f8e5#diff-9341369a460f3f5c5336176dfbca5fbbR10

shaggy

Comment (almogdepaz) [reply cmd/node/node.go:388]: my only point was that this sort of thing can take up lots of time if configured wrong, 
we have enough args as it is we don't need to add keep duplicates.
i opened #798.
anyway that commit is referencing dolev strong its old and not used

**Review by antonlerner — APPROVED** [review]

**Pull request merged**